File: apps/ios/AnpusFeather/Game/Narrative.swift
import CoreGraphics

/// Controls the sequence of scenes — the port of narrative.pde.
enum Mode: Equatable {
    case intro, questions, light, heavy, scales, wisdom, outro
}

final class Narrative {
    private(set) var mode: Mode = .intro
    private var sceneTime: CGFloat = 0

    func update(_ dt: CGFloat) { sceneTime += dt }
    var time: CGFloat { sceneTime }

    func setMode(_ m: Mode) {
        mode = m
        sceneTime = 0
    }
    func resetTime() { sceneTime = 0 }
}

File: apps/unity/Assets/AnpusFeather/Scripts/Narrative.cs
namespace AnpusFeather
{
    /// Controls the sequence of scenes — the port of narrative.pde.
    public enum Mode
    {
        Intro,
        Questions,
        Light,
        Heavy,
        Scales,
        Wisdom,
        Outro
    }

    public sealed class Narrative
    {
        public Mode CurrentMode { get; private set; } = Mode.Intro;
        float _sceneTime;

        public void Update(float dt) => _sceneTime += dt;
        public float Time => _sceneTime;

        public void SetMode(Mode mode)
        {
            CurrentMode = mode;
            _sceneTime = 0f;
        }

        public void ResetTime() => _sceneTime = 0f;
    }
}
